worker/pool.go

```diff
@@ -1,107 +1,29 @@
 package worker
 
 import (
-	"fmt"
-	"sync"
+	"context"
 
-	"github.com/berachain/offchain-sdk/log"
+	"github.com/alitto/pond"
 )
 
-// Pool represents a pool of workers.
-type Pool interface {
-	Start()
-	Stop()
-	AddTask(Executor)
-	RespChan() chan Resultor
-}
-
-// pool is a pool of workers.
-type pool struct {
-	name    string
-	logger  log.Logger
-	execCh  chan Executor
-	resCh   chan Resultor
-	workers []*worker
-	wg      *sync.WaitGroup
-}
-
-// NewPool creates a new pool of workers.
-func NewPool(
-	name string,
-	totalWorkers uint32,
-	logger log.Logger,
-) Pool {
-	// Intialize the pool.
-	p := &pool{
-		name:    name,
-		logger:  logger,
-		execCh:  make(chan Executor),
-		resCh:   make(chan Resultor),
-		workers: make([]*worker, 0),
-		wg:      &sync.WaitGroup{},
-	}
-
-	// Iterate through the number of workers and create them.
-	for i := uint32(0); i < totalWorkers; i++ {
-		w := newWorker(
-			i,
-			p.execCh,
-			p.resCh,
-			logger,
-			p.wg,
-		)
-		p.workers = append(p.workers, w)
+// and other functionality to the pool.
+type Pool struct {
+	name string
+	*pond.WorkerPool
+}
+
+// NewPool creates a new pool.
+func NewPool(ctx context.Context, cfg *PoolConfig) *Pool {
```

Comment:
doesn't need context

Reply:
have to pass the context down to the pool.